src/app/ReadClient.cpp

```diff
@@ -86,7 +86,7 @@ ReadClient::~ReadClient()
         // This won't be the case if the engine shut down before this destructor was called (in which case, mpImEngine
         // will point to null)
         //
-        if (mpImEngine)
+        if (mpImEngine && mpImEngine->InActiveReadClientList(this))
```

Comment:
Why do we need this check?  If we remove this check _and_ remove the manual removal above, won't things work correctly?

Reply:
since the pointer removal is happening inside read client destructor.

Reply:
just remove the manual removal